src/wordlist.c

```diff
@@ -456,6 +456,44 @@ static MAYBE_INLINE int wbuf_unique(char *line)
 	return 1;
 }
 
+#ifdef HAVE_MMAP
+static int mmap_init(int64_t file_len, int mmap_max)
+{
+	if (mem_map)
+		return 1;
+
+	if (mmap_max && mmap_max >= (file_len >> 20)) {
```

Comment:
Looks like we allow almost an extra MiB on top of the configured limit here. Not a new bug, but we could want to rewrite this check (if you do, then do in a separate commit, please).

Reply:
I assume you mean `if (mmap_max && ((uint64_t)mmap_max << 20) >= file_len)`. I was instinctively avoiding overflow but you are right. The cast fixes that.